src/ossl/rsa.rs

```diff
@@ -152,6 +158,12 @@ fn parse_sig_params(
         CKM_SHA3_512_RSA_PKCS_PSS => (SigAlg::RsaPssSha3_512, true),
         _ => return Err(CKR_MECHANISM_INVALID)?,
     };
+    if cfg!(feature = "no_sha1")
+        && (mech.mechanism == CKM_SHA1_RSA_PKCS
+            || mech.mechanism == CKM_SHA1_RSA_PKCS_PSS)
+    {
+        return Err(CKR_MECHANISM_INVALID)?;
+    }
```

Comment:
i would prefer the simple cfg macros over the compound if-cfg conditions
```suggestion
    #[cfg(feature = "no_sha1")]
    if mech.mechanism == CKM_SHA1_RSA_PKCS
        || mech.mechanism == CKM_SHA1_RSA_PKCS_PSS
    {
        return Err(CKR_MECHANISM_INVALID)?;
    }
```
looking that its not shorter, but looks less complex to me.

Same for similar cases in the other files.

Reply:
I would also prefer to avoid the cfg macro wherever possible.

In this case it will always work fine because the constants used are always defined, but the different between the macro and the attribute is:
- with the macro all branches always compile and are elided at run time
- with the attribute instead what follow the attribute is not compiled at all and can contain code that won't compile when the attribute is not set

Reply:
sorry not runtime, they are still elided in a compilation step, but they will fail if they contain reference to variables, functions or constants, etc... that are not available.